modules/ml/src/main/java/org/apache/ignite/ml/math/VectorUtils.java

```diff
@@ -39,4 +39,62 @@ public static DenseLocalOnHeapVector zeroes(int n) {
     public static Vector fromMap(Map<Integer, Double> val, boolean cp) {
         return new MapWrapperVector(val);
     }
+
+    /**
+     * Turn number into Vector of given size with one-hot encoding.
+     *
+     * @param num Number to turn into vector.
+     * @param vecSize Vector size of output vector.
+     * @return One-hot encoded number.
+     */
+    public static Vector num2Vec(int num, int vecSize) {
+        return new DenseLocalOnHeapVector(vecSize).setX(num, 1);
+    }
+
+    /**
+     * Turn Vector into number by looking at index of maximal element in vector.
+     *
+     * @param vec Vector to be turned into number.
+     * @return Number.
+     */
+    public static double vec2Num(Vector vec) {
+        int max = 0;
+        double maxVal = Double.NEGATIVE_INFINITY;
+
+        for (int i = 0; i < vec.size(); i++) {
+            double curVal = vec.getX(i);
+            if (curVal > maxVal) {
+                max = i;
+                maxVal = curVal;
+            }
+        }
+
+        return max;
+    }
+
+    /**
+     * Performs in-place vector multiplication.
+     *
+     * @param vec1 Operand to be changed and first multiplication operand.
+     * @param vec2 Second multiplication operand.
+     * @return Updated first operand.
+     */
+    public static Vector elementWiseTimes(Vector vec1, Vector vec2) {
```

Comment:
All these methods should be placed in Vector classm not in Util

Reply:
Maybe you are right, but I opted not to change public Vector API in this ticket.